Libraries/TranscriptTestRunner/TestRunner.cs

```diff
@@ -201,6 +212,58 @@ protected virtual Task AssertActivityAsync(TestScriptItem expectedActivity, Acti
             return Task.CompletedTask;
         }
 
+        /// <summary>
+        /// Validates the variable date in the bots message with the value between double curly braces.
+        /// </summary>
+        /// <param name="value">The assertion containing the variable.</param>
+        /// <param name="actualActivity">The activity with the message containing the date.</param>
+        protected void ValidateVariable(string value, Activity actualActivity)
+        {
+            var dateRegex = new Regex(@"(\d{1,4}([.\-/])\d{1,2}([.\-/])\d{1,4})");
+            var wordRegex = new Regex(@"[\w]+");
+
+            var dateMatch = dateRegex.Match(actualActivity.Text);
+            var resultExpression = string.Empty;
+            var expectedExpression = wordRegex.Match(value).Value;
+            var dateValue = string.Empty;
+            
+            if (dateMatch.Success)
+            {
+                dateValue = dateMatch.Value;
+                var date = Convert.ToDateTime(dateMatch.Value, CultureInfo.InvariantCulture);
+                resultExpression = EvaluateDate(date);
+            }
+
+            if (resultExpression != expectedExpression)
+            {
+                throw new Exception($"Assertion failed. The variable '{expectedExpression}' does not match with the value {dateValue}.");
+            }
+
+            actualActivity.Text = actualActivity.Text.Replace(dateMatch.Value, value);
+        }
+
+        private static string EvaluateDate(DateTime date)
+        {
+            var currentDate = DateTime.Now.ToString("yyyy-MM-ddTHH:mm:ss.fffZ", CultureInfo.InvariantCulture);
+            var inputDate = date.ToString("yyyy-MM-ddTHH:mm:ss.fffZ", CultureInfo.InvariantCulture);
+            var expression = $"dateReadBack('{currentDate}', '{inputDate}')";
+            var parsed = Expression.Parse(expression);
+
+            if (parsed == null)
+            {
+                throw new Exception("Null parsed expression");
```

Comment:
Consider replacing by:
```suggestion
                throw new InvalidOperationException($"Unable to parse expression: \"{expression}\".");
```

Reply:
Done. Thanks!